src/recorder/stream_recorder/stream_recorder_impl.cpp

```diff
@@ -51,6 +51,13 @@ void StreamRecorderImpl::Run() noexcept try {
 void StreamRecorderImpl::Poll() noexcept {
   if (exit_) {
     try {
+      // Attempt to flush any pending spans before shutting down.
+      //
+      // Note: Flush is non-blocking so we can do this without causing an
+      // unacceptable delay to process termination.
+      Flush();
```

Comment:
Is there any sense in waiting _briefly_, or is there some other, better way for a user to flush before exit?

Reply:
If the user wants to add a wait, they can call the FlushWithTimeout function on the tracer, but I wouldn't force any waiting here.

Most of the time you should be able to write everything, though, without needing a wait.

The effect of this change is that it flushes all the pending data it can without blocking and only drops spans on termination if it doesn't have writable satellite connections